tests/Aspire.Hosting.NodeJs.Tests/NodeJsPublicApiTests.cs

```diff
@@ -0,0 +1,140 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace Aspire.Hosting.NodeJs.Tests;
+
+public class NodeJsPublicApiTests
+{
+    [Fact]
+    public void AddNodeAppShouldThrowWhenBuilderIsNull()
+    {
+        IDistributedApplicationBuilder builder = null!;
+        var name = "nodeapp";
+        var scriptPath = ".\\app.js";
+
+        var action = () => builder.AddNodeApp(name: name, scriptPath: scriptPath);
+
+        var exception = Assert.Throws<ArgumentNullException>(action);
+        Assert.Equal(nameof(builder), exception.ParamName);
+    }
+
+    [Fact]
+    public void AddNodeAppShouldThrowWhenNameIsNull()
+    {
+        var builder = DistributedApplication.CreateBuilder();
+        string name = null!;
+        var scriptPath = ".\\app.js";
+
+        var action = () => builder.AddNodeApp(name: name, scriptPath: scriptPath);
+
+        var exception = Assert.Throws<ArgumentNullException>(action);
+        Assert.Equal(nameof(name), exception.ParamName);
+    }
+
+    [Fact]
+    public void AddNodeAppShouldThrowWhenScriptPathIsNull()
+    {
+        var builder = DistributedApplication.CreateBuilder();
+        var name = "nodeapp";
+        string scriptPath = null!;
+
+        var action = () => builder.AddNodeApp(name: name, scriptPath: scriptPath);
+
+        var exception = Assert.Throws<ArgumentNullException>(action);
+        Assert.Equal(nameof(scriptPath), exception.ParamName);
+    }
+
+    [Fact]
+    public void AddNpmAppShouldThrowWhenBuilderIsNull()
+    {
+        IDistributedApplicationBuilder builder = null!;
+        var name = "npmapp";
+        var workingDirectory = ".\\app";
+
+        var action = () => builder.AddNpmApp(name: name, workingDirectory: workingDirectory);
+
+        var exception = Assert.Throws<ArgumentNullException>(action);
+        Assert.Equal(nameof(builder), exception.ParamName);
+    }
+
+    [Fact]
+    public void AddNpmAppShouldThrowWhenWorkingDirectoryIsNull()
+    {
+        var builder = DistributedApplication.CreateBuilder();
+        var name = "npmapp";
+        string workingDirectory = null!;
+
+        var action = () => builder.AddNpmApp(name: name, workingDirectory: workingDirectory);
+
+        var exception = Assert.Throws<ArgumentNullException>(action);
+        Assert.Equal(nameof(workingDirectory), exception.ParamName);
+    }
+
+    [Fact]
+    public void AddNpmAppShouldThrowWhenNameIsNull()
+    {
+        var builder = DistributedApplication.CreateBuilder();
+        string name = null!;
+        var workingDirectory = ".\\app";
+
+        var action = () => builder.AddNpmApp(name: name, workingDirectory: workingDirectory);
+
+        var exception = Assert.Throws<ArgumentNullException>(action);
+        Assert.Equal(nameof(name), exception.ParamName);
+    }
+
+    [Fact]
+    public void AddNpmAppShouldThrowWhenScriptNameIsNull()
```

Comment:
@Alirexaa you missed this one. It has a default value but it doesn't mean we still can't call it with a `null` value.

Reply:
Also I had to force push to revert the changes in the .sln file.